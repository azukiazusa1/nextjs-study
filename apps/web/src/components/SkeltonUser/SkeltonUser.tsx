import React from 'react';

const SkeltonUser: React.FC = () => {
  return (
    <div className="animate-pulse flex max-w-60">
      <div className="rounded-full bg-slate-700 h-12 w-12" />
      <div className="flex-1 space-y-4 mt-2 ml-4">
        <div className="h-2 bg-slate-700 rounded" />
        <div className="h-2 bg-slate-700 rounded" />
      </div>
    </div>
  );
};

export default SkeltonUser;
