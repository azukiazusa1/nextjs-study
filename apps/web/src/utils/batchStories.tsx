import { composeStories } from '@storybook/testing-react';
import { StoryFile } from '@storybook/testing-react/dist/types';
import { render } from '@testing-library/react';
import React from 'react';

export const batchStories = (stories: StoryFile) => {
  const testCases = Object.values(composeStories(stories)).map((Story: any) => [
    Story.storyName,
    Story,
  ]);

  test.each(testCases)('Renders %s story', async (_storyName, Story) => {
    const tree = await render(<Story />);
    expect(tree.baseElement).toMatchSnapshot();
  });
};
