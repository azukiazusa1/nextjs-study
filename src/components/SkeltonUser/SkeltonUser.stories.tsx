import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';

import SkeltonUser from './SkeltonUser';

export default {
  component: SkeltonUser,
} as ComponentMeta<typeof SkeltonUser>;

const Template: ComponentStory<typeof SkeltonUser> = (args) => <SkeltonUser {...args} />;

export const Default = Template.bind({});
