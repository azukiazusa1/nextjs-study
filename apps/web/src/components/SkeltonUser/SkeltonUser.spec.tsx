import { batchStories } from '@/utils/batchStories';

import * as stories from './SkeltonUser.stories';

describe('components/SkeltonUser', () => {
  batchStories(stories);
});
