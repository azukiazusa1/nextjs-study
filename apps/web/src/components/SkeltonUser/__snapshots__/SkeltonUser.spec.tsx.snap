// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SkeltonUser Renders Default story 1`] = `
<body>
  <div>
    <div
      class="animate-pulse flex max-w-60"
    >
      <div
        class="rounded-full bg-slate-700 h-12 w-12"
      />
      <div
        class="flex-1 space-y-4 mt-2 ml-4"
      >
        <div
          class="h-2 bg-slate-700 rounded"
        />
        <div
          class="h-2 bg-slate-700 rounded"
        />
      </div>
    </div>
  </div>
</body>
`;
